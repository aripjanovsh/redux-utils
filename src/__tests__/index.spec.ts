import {
  ReducerBuilder,
  createAsyncRequestReducer,
  createAsyncValueDictReducer,
  createAsyncValueReducer,
  createFulfillAction,
  createPerformAction,
  createRejectAction,
  getInitialAsyncRequest,
  getInitialAsyncValue,
  isFulfillAction,
  isPerformAction,
  isRejectAction,
  mapToAction,
} from "../index";

describe("index", () => {
  test("public api", () => {
    expect(typeof isPerformAction).toBe("function");
    expect(typeof createPerformAction).toBe("function");
    expect(typeof isFulfillAction).toBe("function");
    expect(typeof createFulfillAction).toBe("function");
    expect(typeof isRejectAction).toBe("function");
    expect(typeof createRejectAction).toBe("function");

    expect(typeof getInitialAsyncRequest).toBe("function");
    expect(typeof createAsyncRequestReducer).toBe("function");

    expect(typeof getInitialAsyncValue).toBe("function");
    expect(typeof createAsyncValueReducer).toBe("function");
    expect(typeof createAsyncValueDictReducer).toBe("function");

    expect(typeof mapToAction).toBe("function");

    expect(typeof ReducerBuilder).toBe("function");
  });
});
