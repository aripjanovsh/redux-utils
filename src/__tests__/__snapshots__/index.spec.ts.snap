// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index public api 1`] = `
Object {
  "AsyncRequest": undefined,
  "AsyncValue": undefined,
  "Dict": undefined,
  "FulfillAction": undefined,
  "PerformAction": undefined,
  "PersistConfigBuilder": [Function],
  "ReducerBuilder": [Function],
  "RejectAction": undefined,
  "SubReducer": undefined,
  "TransformConfig": undefined,
  "composeTransforms": [Function],
  "createAsyncRequestReducer": [Function],
  "createAsyncValueDictReducer": [Function],
  "createAsyncValueReducer": [Function],
  "createExpireTransformConfig": [Function],
  "createFulfillAction": [Function],
  "createPerformAction": [Function],
  "createRejectAction": [Function],
  "createVersionTransformConfig": [Function],
  "getAsyncValueError": [Function],
  "getAsyncValuePayload": [Function],
  "getInitialAsyncRequest": [Function],
  "getInitialAsyncValue": [Function],
  "isAsyncValueFetching": [Function],
  "isFulfillAction": [Function],
  "isPerformAction": [Function],
  "isRejectAction": [Function],
  "mapToAction": [Function],
}
`;
